repos/system_upgrade/el7toel8/actors/peseventsscanner/libraries/library.py

```diff
@@ -0,0 +1,159 @@
+
+
+def parse_action(action_id):
+    """ Get action label from PES event's action id """
+    labels = ('Present', 'Removed', 'Deprecated', 'Replaced', 'Split', 'Merged', 'Moved', 'Renamed')
+
+    if action_id < 0 or action_id >= len(labels):
+        raise ValueError
+
+    return labels[action_id]
+
+
+def parse_packageset(packageset):
+    """ Get packages and repositories data from PES event's package set """
+    if not packageset:
+        return {}
+
+    return {p['name']: p['repository'].lower() for p in packageset.get('package', [])}
+
+
+def parse_entry(entry):
+    """ Parse PES event data """
+    action = parse_action(entry['action'])
+    in_pkgs = parse_packageset(entry.get('in_packageset', {}))
+    out_pkgs = parse_packageset(entry.get('out_packageset', {}))
+
+    return Event(action, in_pkgs, out_pkgs)
+
+
+def parse_file(path):
+    """ Parse JSON file returning PES events """
+    events = []
+    with open(path) as f:
+        data = json.load(f)
+
+        if not isinstance(data, dict):
+            raise ValueError
+
+        for entry in data['packageinfo']:
+            events.append(parse_entry(entry))
+
+        return events
+
+
+def report_data_error():
+    """ Notify about unexpected error while parsing JSON file containing PES events """
+    raise StopActorExecutionError(
+        message='Found outdated PES data.',
+        details={'hint': 'Consult docs to retrieve updated data.'})
+
+
+def filter_events(events):
+    """ Filter PES events """
+    installed_pkgs = set()
+    for pkgs in api.consume(InstalledRedHatSignedRPM):
+        installed_pkgs.update([pkg.name for pkg in pkgs.items])
+
+    filtered = []
+    for event in events:
+        if not installed_pkgs.intersection(set(event.in_pkgs.keys())):
+            continue
+
+        filtered.append(event)
+
+    return filtered
+
+
+def notify_skipped_packages(msg, pkgs):
+    """ Show message about skipped packages """
+    msgs = []
+    msgs.append('{} {}'.format(len(pkgs), msg))
+    msgs.append('\n'.join(['- ' + p for p in pkgs]))
+    api.show_message('\n'.join(msgs))
+
+
+def filter_by_repositories(to_install):
+    """ Filter packages to be installed based on repositories"""
+    blacklisted_pkgs = set()
+    for pkg, repo in to_install.items():
+        if repo in REPOSITORIES_BLACKLIST:
+            blacklisted_pkgs.add(pkg)
+            del to_install[pkg]
+
+    if blacklisted_pkgs:
+        notify_skipped_packages('packages will not be installed due to blacklisted repositories:',
+                                blacklisted_pkgs)
+
+
+def map_repositories(to_install):
+    """ Map repositories from PES data to RHSM repository id """
+    repo_not_mapped_pkgs = set()
+    for pkg, repo in to_install.items():
+        if repo not in REPOSITORIES_MAPPING:
+            repo_not_mapped_pkgs.add(pkg)
+            del to_install[pkg]
+            continue
+
+        to_install[pkg] = REPOSITORIES_MAPPING[repo]
```

Comment:
I wouldn't do this step here, mapping the repository to the target repository could be done at another place, that is dedicated to this part. This is too much of a random place to decide/define the final name of the repository.

Reply:
If I understood correctly, we agree it should be moved but it will not be handled by this PR, right?